### PR TITLE
added helm chart for vsts

### PR DESCRIPTION
**Pull request by bramvdklinkenberg**
**What this PR does / why we need it**: Simple deployment of a vsts build agent on your kubernetes cluster instead of setting up and managing a vm.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes # N/A

**Special notes for your reviewer**:
/assign @mattfarina

Maintainer @bramvdklinkenberg

### CODE REVIEW

**Comment by bramvdklinkenberg** [comment]
@mattfarina do I need to do something or is it your turn? :smiley:

**Comment by unguiculus** [comment]
Please compare with what you get with `helm create` using Helm 2.8+ and update accordingly.

**Comment by bramvdklinkenberg** [comment]
@unguiculus will check it! see what you mean now.

**Comment by unguiculus** [comment]
I did not mention I had issues. I just want you to apply our current best practices.

**Comment by bramvdklinkenberg** [comment]
@unguiculus , I know. I read it wrong but when I read it again I swa what you ment!

**Comment by bramvdklinkenberg** [comment]
@unguiculus , I changed the chart (structure).

File: stable/vstsagent/Chart.yaml [hunk 1]
```diff
@@ -17,3 +17,5 @@ sources:
 maintainers:
 - name: Bram vd Klinkenberg
   email: bramvandenklinkenberg@gmail.com
+tillerVersion: ">2.8.0"
+kubeVersion: ">1.7.7"
```
Comment (unguiculus) [inline stable/vstsagent/Chart.yaml]: \>1.8

File: stable/vstsagent/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {{- with .Values.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+---
+apiVersion: v1
```
Comment (unguiculus) [inline stable/vstsagent/templates/deployment.yaml]: One file per resource please.

File: stable/vstsagent/values.yaml [hunk 3]
```diff
@@ -1,20 +1,54 @@
...
-      cpu: "200m"
+# Default values for vstsagent.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: microsoft/vsts-agent
+  tag: latest
```
Comment (unguiculus) [inline stable/vstsagent/values.yaml]: Use a stable version.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by bramvdklinkenberg** [comment]
@unguiculus I made changes but haven't changed the tagging.
The latest tag always points at a standard image based on the best supported OS that targets VSTS and includes capabilities enabling many of the built-in VSTS build and release tasks.

**Comment by unguiculus** [comment]
Chart releases should have stable image versions. We require that in our guidelines. Using latest is a bad practice and not suitable for production use cases.

**Comment by bramvdklinkenberg** [comment]
@unguiculus , I changed the tag.

File: stable/vstsagent/values.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: microsoft/vsts-agent
-  tag: latest
+  tag: ubuntu-16.04
```
Comment (unguiculus) [inline stable/vstsagent/values.yaml]: This doesn't seem to be a stable tag either.

**Comment by bramvdklinkenberg** [comment]
@unguiculus , I changed the tag to ubuntu-16.04-standard. 

It is confusing what "image tag" is the stable. Microsoft says this about the images:

> Derived images that are based on the standalone agent images provide a variety of capabilities that enable it to support specific VSTS build and release tasks.
The latest tag always points at a standard image based on the best supported OS that targets VSTS and includes capabilities enabling many of the built-in VSTS build and release tasks.

They have a whole bunch of images and tags that can be used, see https://hub.docker.com/r/microsoft/vsts-agent/. But nowhere am I able to see/find a specific image that Mcirsoft points out as stable. You just have a wide variety of images to pick from.

**Review by unguiculus — APPROVED** [review]

**Comment by bramvdklinkenberg** [comment]
@unguiculus , the test failed because of this error;
"error: missing VSTS_ACCOUNT environment variable"

I guess the test doesn't fill in the parameters that are needed. By default these parameters are empty but the need to be provided. But with dummy/ no parameters the deployment will fail.

```
W0525 05:45:25.636] + **helm install --timeout 600 --name vstsage-10192-1 --namespace presubmit-5245-10192 .**
W0525 05:45:25.636] + tee install_output
I0525 05:45:25.736] /src
I0525 05:45:25.737] 
I0525 05:45:25.737] Checking the Chart version has increased for the chart at stable/vstsagent
I0525 05:45:25.737] Unable to find Chart on master. New chart detected.
I0525 05:45:25.737] No requirements found in stable/vstsagent/charts.
I0525 05:45:25.737] /src/stable/vstsagent /src
**I0525 05:45:25.737] Chart does not provide test values. Testing chart with defaults...**
I0525 05:45:28.115] NAME:   vstsage-10192-1
I0525 05:45:28.278] LAST DEPLOYED: Fri May 25 05:45:27 2018
I0525 05:45:28.278] NAMESPACE: presubmit-5245-10192
```

**Comment by unguiculus** [comment]
You can provide test values for CI: https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values

**Comment by bramvdklinkenberg** [comment]
@unguiculus , the test values (I can use my own vsts account plus access token) to test with will not be visible for others?

**Comment by unguiculus** [comment]
Anything you commit is visible, of course.

**Comment by bramvdklinkenberg** [comment]
Any idea of when the last step will take place?

**Comment by bramvdklinkenberg** [comment]
hi @unguiculus , I am/ was wondering why the request has been put on hold.

**Comment by unguiculus** [comment]
@bramvdklinkenberg I put it on hold due to your comment about access tokens.

**Comment by bramvdklinkenberg** [comment]
@unguiculus the only way to test it is with a real vsts account, acces token en agentpool. they have to be provided or else it won;t work.
for testing purposes i just created a temporary access token to pass the tests aand after it passed I deleted the token in my vsts account. default values will not work.

**Comment by unguiculus** [comment]
This will not work. Anyone may create PR to the chart and testing won't work then. Also, the chart will be retested when the bot tries to merge it.

**Comment by bramvdklinkenberg** [comment]
then I guess there is nothing left to do and close this PR.

**Pull request closed without merging**